### PR TITLE
close file descriptors + log piece handoff to sealing subsystem

### PR DESCRIPTION
**Pull request by aarshkshah1992**
To debug https://github.com/filecoin-project/lotus/issues/7144.

- Log import of a file for an offline deal.
- Log the handing off a Piece to the sealing subsystem to add it to a sector.
- Add some missing file closes.

### CODE REVIEW

File: storagemarket/impl/providerstates/provider_states.go [hunk 1]
```diff
@@ -327,11 +328,13 @@ func HandoffDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal stor
 
 		// Hand the deal off to the process that adds it to a sector
 		var packingErr error
+		log.Infow("handing off deal to sealing subsystem", "pieceCid", deal.Proposal.PieceCID, "proposalCid", deal.ProposalCid)
```
Comment (aarshkshah1992) [inline storagemarket/impl/providerstates/provider_states.go:331]: @raulk All of these lines will be ONLY once per deal hence the Info level.

File: storagemarket/impl/provider.go [hunk 2]
```diff
@@ -289,16 +289,19 @@ func (p *Provider) ImportDataForDeal(ctx context.Context, propCid cid.Cid, data
 	if err != nil {
 		return xerrors.Errorf("failed to create temp file for data import: %w", err)
 	}
+	defer tempfi.Close()
```
Comment (nonsense) [inline storagemarket/impl/provider.go:292]: We have that in `cleanup`.

File: storagemarket/impl/provider.go [hunk 3]
```diff
@@ -342,6 +347,10 @@ func (p *Provider) ImportDataForDeal(ctx context.Context, propCid cid.Cid, data
 		return xerrors.Errorf("given data does not match expected commP (got: %s, expected %s)", pieceCid, d.Proposal.PieceCID)
 	}
 
+	log.Debugw("will fire ProviderEventVerifiedData for imported file", "propCid", propCid)
+	if err := tempfi.Close(); err != nil {
+		log.Errorw("failed to close temp imported local file", "propCid", propCid, "err", err)
+	}
```
Comment (nonsense) [inline storagemarket/impl/provider.go:353]: Maybe move this in `cleanup` ?

File: storagemarket/impl/provider.go (hunk 2 above)
Comment (aarshkshah1992) [reply storagemarket/impl/provider.go:292]: `cleanup` is only called if there's some error.

File: storagemarket/impl/providerstates/provider_states.go [hunk 4]
```diff
@@ -311,7 +311,12 @@ func HandoffDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal stor
 		carFilePath = string(file.OsPath())
 
 		// Hand the deal off to the process that adds it to a sector
+		log.Infow("handing off deal to sealing subsystem", "pieceCid", deal.Proposal.PieceCID, "proposalCid", deal.ProposalCid)
 		packingInfo, err = handoffDeal(ctx.Context(), environment, deal, file, uint64(file.Size()))
+		if err := file.Close(); err != nil {
+			log.Errorw("failed to close imported CAR file", "pieceCid", deal.Proposal.PieceCID, "proposalCid", deal.ProposalCid, "err", err)
```
Comment (aarshkshah1992) [inline storagemarket/impl/providerstates/provider_states.go:317]: @raulk @nonsense This is concerning. We never closed the file stream we were handing off to the Miner.

File: storagemarket/impl/providerstates/provider_states.go [hunk 5]
```diff
@@ -311,7 +311,12 @@ func HandoffDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal stor
 		carFilePath = string(file.OsPath())
 
 		// Hand the deal off to the process that adds it to a sector
+		log.Infow("handing off deal to sealing subsystem", "pieceCid", deal.Proposal.PieceCID, "proposalCid", deal.ProposalCid)
 		packingInfo, err = handoffDeal(ctx.Context(), environment, deal, file, uint64(file.Size()))
+		if err := file.Close(); err != nil {
```
Comment (aarshkshah1992) [inline storagemarket/impl/providerstates/provider_states.go:316]: @raulk @nonsense This is concerning. We never closed the file stream we were handing off to the Miner.

File: storagemarket/impl/provider.go (hunk 2 above)
Comment (nonsense) [reply storagemarket/impl/provider.go:292]: Right, I was confused about that. It seems like it is missing on line 339 though.

File: storagemarket/impl/provider.go [hunk 6]
```diff
@@ -342,6 +347,10 @@ func (p *Provider) ImportDataForDeal(ctx context.Context, propCid cid.Cid, data
...
 	return p.deals.Send(propCid, storagemarket.ProviderEventVerifiedData, tempfi.Path(), filestore.Path(""))
```
Comment (raulk) [inline storagemarket/impl/provider.go:354]: What is this empty filestore path? Is this expected? (I know this PR hasn't touched this)

File: storagemarket/impl/provider.go (hunk 3 above)
Comment (raulk) [reply storagemarket/impl/provider.go:353]: It looks redundant to both defer a close and close explicitly here. The deferred close should be enough.

File: storagemarket/impl/providerstates/provider_states.go [hunk 7]
```diff
@@ -327,11 +332,13 @@ func HandoffDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal stor
 
 		// Hand the deal off to the process that adds it to a sector
 		var packingErr error
+		log.Infow("handing off deal to sealing subsystem", "pieceCid", deal.Proposal.PieceCID, "proposalCid", deal.ProposalCid)
 		packingInfo, packingErr = handoffDeal(ctx.Context(), environment, deal, v2r.DataReader(), v2r.Header.DataSize)
 		// Close the reader as we're done reading from it.
 		if err := v2r.Close(); err != nil {
 			return ctx.Trigger(storagemarket.ProviderEventDealHandoffFailed, xerrors.Errorf("failed to close CARv2 reader: %w", err))
 		}
+		log.Infow("closed CARv1 reader after handing off deal to sealing subsystem", "pieceCid", deal.Proposal.PieceCID, "proposalCid", deal.ProposalCid)
```
Comment (raulk) [inline storagemarket/impl/providerstates/provider_states.go:341]:
```suggestion
		log.Infow("closed car datareader after handing off deal to sealing subsystem", "pieceCid", deal.Proposal.PieceCID, "proposalCid", deal.ProposalCid)
```

File: storagemarket/impl/provider.go (hunk 2 above)
Comment (raulk) [reply storagemarket/impl/provider.go:292]: True.

File: storagemarket/impl/provider.go (hunk 6 above)
Comment (aarshkshah1992) [reply storagemarket/impl/provider.go:354]: Hey , that's the "metdata -path", a remanent of the universal retrieval era.

File: storagemarket/impl/provider.go (hunk 2 above)
Comment (aarshkshah1992) [reply storagemarket/impl/provider.go:292]: Added missing cleanup call.

**Review by nonsense — APPROVED** [review]

File: storagemarket/impl/provider.go (hunk 3 above)
Comment (aarshkshah1992) [reply storagemarket/impl/provider.go]: Removed redundant close.

**Review by raulk — APPROVED** [review]

**Pull request merged**